src/codegen/codegen_c_visitor.cpp

```diff
@@ -1418,7 +1418,15 @@ void CodegenCVisitor::print_function_call(const FunctionCall& node) {
             printer->add_text(", ");
         }
     } else if (nmodl::details::needs_neuron_thread_first_arg(function_name) &&
-               arguments.front()->get_node_name() != "nt") {
+               (arguments.front()->get_node_type() != AstNodeType::NAME &&
+                    arguments.front()->get_node_type() != AstNodeType::STRING &&
+                    arguments.front()->get_node_type() != AstNodeType::CONSTANT_VAR &&
+                    arguments.front()->get_node_type() != AstNodeType::VAR_NAME &&
+                    arguments.front()->get_node_type() != AstNodeType::LOCAL_VAR ||
+                arguments.front()->get_node_name() != "nt")) {
```

Comment:
```suggestion
               ((arguments.front()->get_node_type() == AstNodeType::NAME ||
                    arguments.front()->get_node_type() == AstNodeType::STRING ||
                    arguments.front()->get_node_type() == AstNodeType::CONSTANT_VAR ||
                    arguments.front()->get_node_type() == AstNodeType::VAR_NAME ||
                    arguments.front()->get_node_type() == AstNodeType::LOCAL_VAR) &&
                arguments.front()->get_node_name() != "nt")) {
```
I think this is a bit more understandable like this?
Also maybe you could set all the checks for the `AstNodeType`s in a variable to make it more clear?
```
auto possible_nt_variable_types = arguments.front()->get_node_type() == AstNodeType::NAME ||
                    arguments.front()->get_node_type() == AstNodeType::STRING ||
                    arguments.front()->get_node_type() == AstNodeType::CONSTANT_VAR ||
                    arguments.front()->get_node_type() == AstNodeType::VAR_NAME ||
                    arguments.front()->get_node_type() == AstNodeType::LOCAL_VAR;
```